geometry/cbox.cpp

```diff
@@ -0,0 +1,142 @@
+#include "geometry/cbox.hpp"
+
+#include "geometry/bbox.hpp"
+#include "geometry/convex_hull.hpp"
+#include "geometry/line2d.hpp"
+#include "geometry/polygon.hpp"
+#include "geometry/segment2d.hpp"
+
+#include "base/assert.hpp"
+#include "base/logging.hpp"
+
+#include <algorithm>
+#include <array>
+#include <limits>
+
+using namespace std;
+
+namespace m2
+{
+namespace
+{
+static_assert(numeric_limits<double>::has_infinity, "");
+
+double const kInf = numeric_limits<double>::infinity();
+double const kEps = 1e-12;
+
+// Checks whether (p1 - p) x (p2 - p) >= 0.
+bool IsCCW(PointD const & p1, PointD const & p2, PointD const & p)
+{
+  return robust::OrientedS(p1, p2, p) > -kEps;
+}
+
+PointD Ort(PointD const & p) { return PointD(-p.y, p.x); }
+
+template <typename Fn>
+void ForEachRect(ConvexHull const & hull, Fn && fn)
```

Comment:
Please provide a short description of this function.

Reply:
Done